version-detection/version-detection.md

```diff
@@ -0,0 +1,124 @@
+
+This section defines the "text/gedcom" media type for use with protocols such as HTTP or email, to identify content
+as being a GEDCOM file.  The Internet Assigned Numbers Authority (IANA) maintains the
+[registry of assigned media types](https://www.iana.org/assignments/media-types/media-types.xhtml).
+
+The following registration for "text/gedcom" is specified using the template defined in
+[RFC 4288](https://www.rfc-editor.org/rfc/rfc4288).
+
+**Name**: text/gedcom
+
+**Parameters**:
+
+* **charset**: As specified in [RFC 2046, section 4.1.2](https://www.rfc-editor.org/rfc/rfc2046.html#section-4.1.2).
+  The parameter is not used when payload is present because the charset information is transported inside
+  the payload.  However, should charset negotiation be needed, and no Accept-Charset header or equivalent
+  is supported, it may be used in a media range used in a request, such as in an Accept header.
+
+  Many earlier versions of GEDCOM specified an "ANSEL" character set, which does not have an assignment in 
+  the [IANA Character Sets registry](https://www.iana.org/assignments/character-sets/character-sets.xhtml)
+  and so cannot be used by either the charset parameter or Accept-Charset.  Since UTF-16 support was added
+  in 1993, UTF-8 has been supported in GEDCOM since 1999, and the latest version of GEDCOM only supports UTF-8,
+  it is not expected that ANSEL would need to be requested.
+
+* **version**: The GEDCOM specification version.  The parameter is not used when payload is present because
+  the charset information is transported inside the payload.  However, should version negotiation be needed,
+  it may be used in a media range used in a request, such as in an Accept header.
+
+**Encoding considerations**: Text in a specified character set.
+
+**Security considerations**:
+Can contain directives to read files on the local hard drive and send GET requests to HTTP and HTTPS URLs.
+No active content, no file modification directives, no information-sharing directives.
+Commonly used to store genealogical information, which may include personal and sensitive information as well
+as information commonly used to create security questions. This information is not encrypted or otherwise
+protected by GEDCOM itself; hence, it falls on the software and persons managing the files to ensure they are
+kept confidential if they contain such information.
+
+**Interoperability considerations**: The GEDCOM version is encoded in the content, and an algorithm is provided
+for detecting the version.
+
+**Application usage**: GEDCOM is used by family history and related applications, including family trees,
+one-place and one-name studies, and historical analysis of interrelated individuals.
+
+**Fragment Identifier**: not used
+
+**Restrictions on Use**: none
+
+**Provisional Registration**: This media type is intended to be permanent.
+
+**Magic numbers**: One of the following sequences of bytes should be present at the start of the file:
```

Comment:
These magic numbers are not _sufficient_ to determine whether a file with a .ged extension is indeed a GEDCOM file.

To be a valid GEDCOM file, `0 HEAD` must be immediately followed by a valid line terminator.

A check for a valid line terminator is needed to ensure distinguish between other files which use the .ged extension with the same magic numbers but which are not GEDCOM files.  By far the most common source of such files is some versions of "Family Tree Maker for Windows" which allows users to produce "GEDCOM" files beginning `0 HEADER`.

However, any check for the presence of a line terminator should be forgiving.  Many applications erroneously export an additional space between the `HEAD` tag the line terminator and such cases should be handled.